src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs

```diff
@@ -92,7 +112,17 @@ public TypeBuilder DefineType(string name, TypeAttributes attr)
 
         public TypeBuilder DefineType(string name, TypeAttributes attr,
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? parent, PackingSize packingSize, int typesize)
-                => DefineTypeCore(name, attr, parent, null, packingSize, typesize);
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
+
+            if (name[0] == '\0')
+                throw new ArgumentException(SR.Argument_IllegalName, nameof(name));
+
+            if (name.Length > 1023)
+                throw new ArgumentException(SR.Argument_TypeNameTooLong, nameof(name));
```

Comment:
```suggestion
            if (name[0] == '\0')
                throw new ArgumentException(SR.Argument_IllegalName, nameof(name));

            if (name.Length > 1023)
                throw new ArgumentException(SR.Argument_TypeNameTooLong, nameof(name));
```
I would suggest to keep this validation in the existing runtime builders, and omit it in the managed builder. Nothing in the ECMA spec says that these names are invalid. It is arbitrary limitation, potentially compensating for bugs elsewhere.

I think that the only validation that makes sense here is `ArgumentException.ThrowIfNullOrEmpty`.

Reply:
(This comment is for all places with this pattern.)